explorer/common/trace_stream.h

```diff
@@ -23,27 +43,55 @@ namespace Carbon {
 // is fully treated as a separate file, we should be able to take a different
 // approach where the caller explicitly toggles tracing when switching file
 // contexts.
+
 class TraceStream {
  public:
   // Returns true if tracing is currently enabled.
   auto is_enabled() const -> bool {
-    return stream_.has_value() && !in_prelude_;
+    return stream_.has_value() &&
+           current_code_context_ != CodeContext::Prelude &&
+           (allowed_phases_[current_phase_] ||
+            allowed_phases_[ProgramPhase::All]);
```

Comment:
Rather than treating `All` as special here, I wonder if we could make `set_allowed_phases({All})` mark all the phases as allowed? (`allowed_phases_.set()`)